aea/helpers/search/models.py

```diff
@@ -918,32 +1379,30 @@ def __str__(self):
             [str(c) for c in self.constraints], self.model
         )
 
-    @classmethod
-    def encode(cls, query_protobuf_object, query_object: "Query") -> None:
+    def encode(self) -> models_pb2.Query.Model():
         """
-        Encode an instance of this class into the protocol buffer object.
-
-        The protocol buffer object in the query_protobuf_object argument must be matched with the instance of this class in the 'query_object' argument.
+        Encode an instance of this class into a protocol buffer object.
 
-        :param query_protobuf_object: the protocol buffer object whose type corresponds with this class.
-        :param query_object: an instance of this class to be encoded in the protocol buffer object.
-        :return: None
+        :return: the matching protocol buffer object
         """
-        query_bytes = pickle.dumps(query_object)  # nosec
-        query_protobuf_object.query_bytes = query_bytes
+        query = models_pb2.Query.Model()
+        constraint_expr_pbs = [ConstraintExpr._encode(constraint) for constraint in self.constraints]
+        query.constraints.extend(constraint_expr_pbs)
+
+        if self.model is not None:
+            query.model.CopyFrom(self.model.encode())
+        return query
 
     @classmethod
     def decode(cls, query_protobuf_object) -> "Query":
         """
         Decode a protocol buffer object that corresponds with this class into an instance of this class.
 
-        A new instance of this class must be created that matches the protocol buffer object in the 'query_protobuf_object' argument.
-
-        :param query_protobuf_object: the protocol buffer object whose type corresponds with this class.
-        :return: A new instance of this class that matches the protocol buffer object in the 'query_protobuf_object' argument.
+        :param query_protobuf_object: the protocol buffer object corresponding with this class.
+        :return: A new instance of this class matching the protocol buffer object
         """
-        query = pickle.loads(query_protobuf_object.query_bytes)  # nosec
-        return query
+        constraints = [ConstraintExpr._decode(c) for c in query_protobuf_object.constraints]
+        return cls(constraints, DataModel.decode(query_protobuf_object.model) if query_protobuf_object.HasField("model") else None)
```

Comment:
simpler to read if the second arg is first declared in the line above

Reply:
Done